projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-saving-post/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { usePrevious } from '@wordpress/compose';
+import { store as editorStore } from '@wordpress/editor';
+import { useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * React hook to detect when the post is just saved,
+ * running the callback when it happens.
+ * Additionally, it accepts a dependency array which is passed to useEffect hook.
+ *
+ * @param {Function} fn - Callback function to run when the post is just saved.
+ * @param {Array} deps  - Depencency array.
+ */
+export function usePostJustSaved( fn, deps ) {
+	const isSaving = useSelect( select => select( editorStore ).isSavingPost(), [] );
+	const wasSaving = usePrevious( isSaving );
+
+	useEffect( () => {
+		if ( ! ( wasSaving && ! isSaving ) ) {
+			return;
+		}
+
+		fn();
+	}, [ isSaving, wasSaving, fn, deps ] );
+}
+
+/**
+ * React hook to detect when the post is just published,
+ * running the callback when it happens.
+ * Additionally, it accepts a dependency array which is passed to useEffect hook.
+ *
+ * @param {Function} fn - Callback function to run when the post is just published.
+ * @param {Array} deps  - Depencency array.
+ */
+export function usePostJustPublished( fn, deps ) {
```

Comment:
I wonder if this should live in a separate file? We're in a `use-saving-post` directory, but `useSavingPost` doesn't exist.

Reply:
Or maybe combine the two and have a `usePostJust` hook. Something along the lines of:

```
function usePostJust( state, fn, deps ) {
    const isInState = useSelect( select => select( editorStore )[ 'saved' === state ? 'isSavingPost' : 'isPublishingPost'](), [] );
    const wasInState = usePrevious( isInState );

   ....
```

I'm not sure if that over-complicates things or simplifies them! Thoughts?

Reply:
why is the issue with defining all hooks in the same file? is it a naming matter?

I can split the hooks into different files. That isn't a problem.